FlashLFQ/Bayesian Protein Quantification/ProteinQuantificationEngine.cs

```diff
@@ -0,0 +1,604 @@
+                                randomSeedsForEachProtein[protein][0], null, false, BurnInSteps, NSteps, out var mus);
+
+                            var skepticalResult = RunBayesianProteinQuant(peptideFoldChanges, protein, condition,
+                                randomSeedsForEachProtein[protein][1], nullHyp, true, BurnInSteps, NSteps, out var skepticalMus);
+
+                            result.cutoff = nullHyp;
+                            result.CalculatePosteriorErrorProbability(skepticalMus);
+                        }
+                        else if (peptideFoldChanges.Count == 1)
+                        {
+                            result = new ProteinQuantificationEngineResult(protein, BaseCondition, condition,
+                                new double[] { peptideFoldChanges.First() }, new double[] { double.NaN },
+                                new double[] { double.NaN }, peptideFoldChanges, ProteinToBaseConditionIntensity[protein]);
+
+                            result.cutoff = nullHyp;
+                            result.CalculatePosteriorErrorProbability(new double[] { nullHyp });
+                        }
+                        else
+                        {
+                            result = new ProteinQuantificationEngineResult(protein, BaseCondition, condition,
+                                new double[] { 0.0 }, new double[] { double.NaN }, new double[] { double.NaN },
+                                peptideFoldChanges, ProteinToBaseConditionIntensity[protein]);
+
+                            result.cutoff = nullHyp;
+                            result.CalculatePosteriorErrorProbability(new double[] { nullHyp });
+                        }
+
+                        protein.conditionToQuantificationResults.Add(condition, result);
+                    }
+                });
+            }
+
+            // calculate FDR for the condition
+            foreach (var condition in conditions)
+            {
+                var res = proteinList.Select(p => p.Key.conditionToQuantificationResults[condition]).ToList();
+                CalculateFalseDiscoveryRates(res);
+            }
+
+            // remove temporary conditions
+            if (!conditionsAreDefined)
+            {
+                foreach (var file in results.SpectraFiles)
+                {
+                    file.Condition = "";
+                }
+            }
+        }
+
+        /// <summary>
+        /// Pairs up proteins with constituent peptides and calculates peptide biorep intensities for easy lookup.
+        /// </summary>
+        private void Setup()
+        {
+            // temporarily set conditions if they aren't defined
+            if (!conditionsAreDefined)
+            {
+                foreach (var file in results.SpectraFiles)
+                {
+                    file.Condition = file.FullFilePathWithExtension;
```

Comment:
This just assigns a placeholder that ensures no files share a condition, which makes comparisons impossible?

Reply:
no - comparisons are made, but just on a one file vs. one file basis

Reply:
I don't understand how comparisons can be made with just two files. There's not a standard deviation at that point, so if the values are not exactly identical (1.0 vs 1.00000001), then they're different...?

Reply:
a comparison can be made if there are multiple peptides per protein. the N isn't the number of bioreps - it's the number of peptide fold-change measurements. each of which is an independent measurement

Reply:
Is that useful, though? It's not really measuring biological variance (which is what we're interested in), but rather how much peptides from a shared protein vary against one another. I'm not feeling super good about that comparison.

Reply:
I think I see the hangup. if, let's say, N=1, a t-test is not possible. In Bayesian statistics, it's possible to estimate a probability of a change (though IMO it's not especially useful, so I require at least N=2). in this method, thousands of t-distributions are generated by Monte Carlo sampling. if you have one datapoint, a wide range of very different distributions seem plausible. thus, there is great uncertainty in the mean. as the number of datapoints increases, and the datapoints become more precise, the MCMC sampling generates t distributions that look very similar, because if it generated a t distribution with a vastly different mean/stdev, the likelihood of that distribution representing the data is very low. thus, the sampled t-distributions have a relatively precise estimate of the mean.